app/scripts/controllers/permissions/selectors.js

```diff
@@ -112,3 +139,122 @@ export const diffMap = (currentMap, previousMap) => {
   }
   return changedMap;
 };
+
+/**
+ * Given the current and previous exposed CAIP-25 authorization for each PermissionController
+ * subject, returns a new map containing the current value of scopes added/changed in an authorization.
+ * The values of each map must be immutable values directly from the
+ * PermissionController state, or an empty object instantiated in this
+ * function.
+ *
+ * @param {Map<string, Caip25Authorization>} newAuthorizationsMap - The new origin:authorization map.
+ * @param {Map<string, Caip25Authorization>} [previousAuthorizationsMap] - The previous origin:authorization map.
+ * @returns {Map<string, Caip25Authorization>} The origin:authorization map of changed authorizations.
+ */
+export const getChangedAuthorizations = (
```

Comment:
This doesn't appear to be a selector 🤔 Perhaps we can move it to another file? It's a bit confusing to keep it here.

Reply:
Same for `getRemovedAuthorizations`

Reply:
Moved to a new `differ.ts` file here https://github.com/MetaMask/metamask-extension/pull/27782/commits/f4b58b4e95dfcaf682d4a7774ad2792edddc5cd7